test/load/allocation/grpc/allocationload.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"sync"
+	"time"
+
+	pb "agones.dev/agones/pkg/allocation/go"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+func main() {
+	keyFile := flag.String("key", "missing key", "the private key file for the client certificate in PEM format")
+	certFile := flag.String("cert", "missing cert", "the public key file for the client certificate in PEM format")
+	cacertFile := flag.String("cacert", "missing cacert", "the CA cert file for server signing certificate in PEM format")
+	externalIP := flag.String("ip", "missing external IP", "the external IP for allocator server")
+	port := flag.String("port", "443", "the port for allocator server")
+	namespace := flag.String("namespace", "default", "the game server kubernetes namespace")
+	multicluster := flag.Bool("multicluster", false, "set to true to enable the multi-cluster allocation")
+	numOfClients := flag.Int("numberofclients", 1, "number of clients to do allocations in parallel")
+	perClientAllocs := flag.Int("perclientallocations", 1, "numbere of allocations to be done per client")
+
+	flag.Parse()
+
+	endpoint := *externalIP + ":" + *port
+	cert, err := ioutil.ReadFile(*certFile)
+	if err != nil {
+		panic(err)
+	}
+	key, err := ioutil.ReadFile(*keyFile)
+	if err != nil {
+		panic(err)
+	}
+	cacert, err := ioutil.ReadFile(*cacertFile)
+	if err != nil {
+		panic(err)
+	}
+
+	request := &pb.AllocationRequest{
+		Namespace: *namespace,
+		MultiClusterSetting: &pb.MultiClusterSetting{
+			Enabled: *multicluster,
+		},
+	}
+
+	dialOpts, err := createRemoteClusterDialOption(cert, key, cacert)
+	if err != nil {
+		panic(err)
+	}
+	conn, err := grpc.Dial(endpoint, dialOpts)
+	if err != nil {
+		panic(err)
+	}
+
+	grpcClient := pb.NewAllocationServiceClient(conn)
```

Comment:
Should the gRPC client be created within the go routing for per client? 
That would be a closer per client simulation.

Reply:
I tried it inside the loop and performance was same. See the update

Reply:
Thanks for the change. You should also add dial to the per client go routine.

Reply:
That reduces the throughput. We can consider different test such as 5 MM client (e.g one per continent) with concurrent allocations using the same client

Reply:
I made the change. With 40 clients, we get few more errors (still reasonable). To allocate 5000 GSs with 40 clients, it takes ~70 secs. To increase throughput, we probably need more allocator service replicas

Reply:
I think if you only use one connection, that connection goes to the same pod, therefore you are not testing multiple clients/pods.

Reply:
> I think if you only use one connection, that connection goes to the same pod, therefore you are not testing multiple clients/pods.

This is an interesting question actually. But a couple of thoughts:

1. I think you are right @pooneh-m - and we want to test across multiple pods. But I'm not sure if that's the requirement for this PR.
1. Do we expect someone is going to have 50 matchmakers? Seems unlikely - so having a single client may actually make sense, since that's what most architectures will have (or at least, a very low number)
1. (Separate to this PR) Should we look at how we can make it easier to better implement gRPC load balancing? Some resources:
    1. https://grpc.io/blog/grpc-load-balancing/
    1. https://blog.nobugware.com/post/2019/kubernetes_mesh_network_load_balancing_grpc_services/
    1. https://kubernetes.io/blog/2018/11/07/grpc-load-balancing-on-kubernetes-without-tears/
    1. https://itnext.io/proxyless-grpc-load-balancing-in-kubernetes-ca1a4797b742

If so, this sounds like something we should track in a separate issue for discussion.

Reply:
I think it is a good discussion.

The current load test client, creates one connection per client so it should leverage all the pods. The performance is more or less so keeping it as is.